mlflow/genai/scorers/builtin_scorers.py

```diff
@@ -1209,6 +1209,191 @@ def __call__(
         return _sanitize_scorer_feedback(feedback)
 
 
+@experimental(version="3.8.0")
+@format_docstring(_MODEL_API_DOC)
+class ConversationalSafety(BuiltInScorer):
+    """
+    Conversational safety evaluates whether an entire multi-turn conversation is safe.
+
+    This scorer analyzes the complete conversation for various safety concerns including:
+
+    - Harmful content (hate speech, harassment, violence, illegal acts)
+    - Jailbreak attempts (users trying to bypass safety guidelines)
+    - Harmful escalation patterns (conversation becoming progressively unsafe)
+    - Manipulation attempts (tricking the assistant into unsafe outputs)
+    - Assistant safety failures (assistant providing harmful responses)
+
+    You can invoke the scorer directly with a single input for testing, or pass it to
+    `mlflow.genai.evaluate` for running full evaluation on a dataset.
+
+    Args:
+        name: The name of the scorer. Defaults to "conversational_safety".
+        model: {{ model }}
+
+    Example (direct usage with messages):
+
+    .. code-block:: python
+
+        import mlflow
+        from mlflow.genai.scorers import ConversationalSafety
+
+        # Evaluate a multi-turn conversation
+        feedback = ConversationalSafety()(
+            messages=[
+                {"role": "user", "content": "What is the capital of France?"},
+                {"role": "assistant", "content": "The capital of France is Paris."},
+                {"role": "user", "content": "Tell me more about it."},
+                {"role": "assistant", "content": "Paris is known for the Eiffel Tower."},
+            ]
+        )
+        print(feedback.value)  # "yes" if safe, "no" if unsafe
```

Comment:
Huh does this work? I had thought that conversational judges required trace input, like in this example: https://github.com/mlflow/mlflow/commit/96f5dc5b623e9092f76314189cb7898af20aac3a#diff-44d007f2c18ee4dab492ea83530436269daf9623b98cbe9700684a9ba9f434b4R1709